src/ui/statemachine/defaulticonstate.js

```diff
@@ -0,0 +1,45 @@
+import { State } from './statemachine';
+import LoadingIconState from './loadingiconstate';
+import DOM from '../dom/dom';
+
+/**
+ * Defines behavior for search bar icon in default state and during state transitions
+ */
+export default class DefaultIconState extends State {
+  constructor (controller, useCustomIcon) {
+    super('default');
+    this._controller = controller;
+    this._useCustomIcon = useCustomIcon;
+    this._defaultIconElement = useCustomIcon
+      ? DOM.query(this._controller.searchBarContainer, '.js-yxt-SearchBar-buttonImage')
+      : DOM.query(this._controller.searchBarContainer, '.js-yxt-AnimatedReverse');
+  }
+
+  canEnter (e) {
+    let focusStillInSearchbar = false;
+    if (e && e.relatedTarget) {
+      focusStillInSearchbar = this._controller.searchBarContainer.contains(e.relatedTarget);
+    }
+    return !this._controller.iconIsFrozen && !focusStillInSearchbar;
+  }
+
+  onEnter (prevState) {
+    this._useCustomIcon ? this._handleTransitionToCustom() : this._handleTransitionToYext(prevState);
+  }
+
+  _handleTransitionToYext (prevState) {
+    this._defaultIconElement.classList.remove('yxt-SearchBar-Icon--inactive');
+    const svgWrapper = DOM.query(this._defaultIconElement, '.Icon--yext_animated_reverse');
```

Comment:
Can we move some of these `DOM.query` calls to the constructor? I think it's a little better for performance if we don't make a DOM query on every transition if it's not needed